### PR TITLE
Validate facility form fields

### PR DESCRIPTION
**Pull request by nickscialli-usds**
## Related Issue or Background Info

- Issue #917

## Changes Proposed

- Adds `yup` validation to facility fields
- Makes sure to validate the _correct_ fields
- Adds same phone number validation on frontend as is on backend (google)

## Additional Information

- This has more boilerplate than I'd ultimately like. I think we could use `react-hook-form` or another established form library on the front-end to make our forms consistent, have less boilerplate, and integrate relatively easily with a validation library like yup.

### CODE REVIEW

**Comment by nickscialli-usds** [comment]
Would like to ignore the SonarCloud duplication failure, but will defer to reviewers on that one.

Basically, I think we have some tech debt in that we don't have a form management library to abstract away my home-rolled `validateField` and `clearError` functions. Therefore, I wrote them and then duplicated them across two components and SonarCloud is not a fan of that.

File: frontend/src/app/Settings/Facility/FacilityFormContainer.tsx [hunk 1]
```diff
@@ -141,7 +139,6 @@ const ADD_FACILITY_MUTATION = gql`
       street: $street
       streetTwo: $streetTwo
       city: $city
-      county: ""
```
Comment (timbest-usds) [inline frontend/src/app/Settings/Facility/FacilityFormContainer.tsx:144]: Thank you for cleaning this up

**Review by timbest-usds — APPROVED** [review]
I support ignoring the sonar duplication warning until we have a better method for managing forms

File: frontend/src/app/Settings/Facility/facilitySchema.ts [hunk 2]
```diff
@@ -0,0 +1,78 @@
...
+>;
+
+const providerSchema: yup.SchemaOf<RequiredProviderFields> = yup.object({
+  firstName: yup.string().required("Ordering provider first name is missing"),
+  middleName: yup.string(),
+  lastName: yup.string().required("Ordering provider last name is missing"),
+  suffix: yup.string(),
+  NPI: yup.string().required("Ordering provider NPI is missing"),
+  phone: yup.string(),
+  street: yup.string().required("Ordering provider street is missing"),
```
Comment (timbest-usds) [inline frontend/src/app/Settings/Facility/facilitySchema.ts:25]: The Ordering Providers address is not required. `street`, `state`, and `zipCode` should be optional

**Review by timbest-usds — CHANGES_REQUESTED** [review]

File: frontend/src/app/Settings/Facility/facilitySchema.ts (hunk 2 above)
Comment (timbest-usds) [reply frontend/src/app/Settings/Facility/facilitySchema.ts:25]: ....turns out nothing about the ordering provider is required

**Review by timbest-usds — APPROVED** [review]

**Pull request merged**